Derpy/Utils/Tumblr/TumblrClient.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Sentry;
+using Sentry.Protocol;
+using Serilog;
+
+namespace Derpy.Utils.Tumblr
+{
+    public class TumblrClient : ITumblrClient
+    {
+        private readonly IWebClient _webClient;
+        private readonly string _apiKey;
+
+        private static Uri apiUrl = new Uri("https://api.tumblr.com/v2/blog/");
+        private static JsonSerializerOptions _serializerOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = new SnakeCaseNamingPolicy()
+        };
+
+        public TumblrClient(IWebClient webClient, IKeyProvider keyProvider)
+        {
+            _webClient = webClient;
+            _apiKey = keyProvider.TumblrApiKey;
+        }
+
+        // https://www.tumblr.com/docs/en/api/v2#posts--retrieve-published-posts
+        // https://github.com/tumblr/docs/blob/master/api.md#posts--retrieve-published-posts
+        public async Task<string[]> GetAllPostUrlsAsync(string blogIdentifier, string tag = null)
+        {
+            var @params = new Dictionary<string, string>
+            {
+                { "api_key", _apiKey }
+            };
+
+            if (!string.IsNullOrWhiteSpace(tag))
+            {
+                @params.Add("tag", tag);
+            }
+
+            var uri = BuildUri(blogIdentifier, "/posts", @params);
+            var response = await _webClient.GetAsync(uri);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Log.Error("Tumblr posts GET response returned {StatusCode}", response.StatusCode);
+                SentrySdk.CaptureMessage($"Tumblr posts GET response returned {response.StatusCode}", SentryLevel.Error);
```

Comment:
We would probably profit from having the full response sent to Sentry here too (possibly in additional attribute).

Also for everything we send to sentry, have an idea of what failed.

Reply:
I'm still pretty new to Sentry, would doing that be something like replacing the current code in the `if` block with this?

```c#
using (SentrySdk.PushScope())
{
    SentrySdk.ConfigureScope(scope =>
    {
        scope.SetExtra("response", response);
    });

    Log.Error("Tumblr posts GET response returned {StatusCode}", response.StatusCode);
    SentrySdk.CaptureMessage($"Tumblr posts GET response returned {response.StatusCode}", SentryLevel.Error);
}
```

Reply:
That looks good, gonna have to test what it gives